components/asset-metadata-service/pkg/processor/processor.go

```diff
@@ -0,0 +1,152 @@
+package processor
```

Comment:
What do you think about creating generic `processor` that will be used for multithreading in whole asset store domain?

Reply:
Yup, that's why I created the package, where you provide worker function. I think it is generic enough to use it somewhere else, but if you don't agree, write your concerns here 🙂